packages/cadl-ranch-specs/http/special-headers/request-id/main.tsp

```diff
@@ -0,0 +1,59 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+@doc("Different request id header configurations.")
+@supportedBy("dpg")
+@scenarioService("/special-headers/request-id")
+namespace SpecialHeaders.RequestId;
+
+@client
+namespace DefaultClient {
+  @scenario
+  @scenarioDoc("""
+Default situation: if no request id header configured, we use standard request id header.
+""")
+  @doc("""
+Default request id header.
+""")
+  @post
+  @route("/default")
+  op default(): NoContentResponse;
+}
+
+@client
+namespace StandardClient {
+  @scenario
+  @scenarioDoc("""
+Standard request id header. SDK should not genreate `clientRequestId` paramerter but use policy to auto-set the header.
+""")
+  @doc("""
+Standard request id header.
+""")
+  @post
+  @route("/standard")
+  op standard(
+    @header("x-ms-client-request-id")
```

Comment:
https://github.com/Azure/typespec-azure/blob/main/packages/typespec-azure-core/lib/traits.tsp#L133

Could you also add a test for azure-core - SupportsClientRequestId?

Reply:
https://github.com/Azure/cadl-ranch/blob/main/packages/cadl-ranch-specs/http/azure/core/traits/main.tsp#L62

Reply:
Thanks for the info, I think we could add a test case for `NoClientRequestId`.

https://github.com/Azure/typespec-azure/blob/main/packages/typespec-azure-core/lib/traits.tsp#L148

Reply:
`NoClientRequestId` is up to you (and maybe other langauges).

Java's logic is "x-ms-client-request-id" is **always** sent, regardless of TSP or Swagger or whatever. The only exception maybe when find this "client-request-id" header in client.
So Java won't test on `NoClientRequestId`, as it won't ever happen in Java SDK.

Reply:
I've removed the standard ones and update the doc for traits test.

Reply:
@weidongxu-microsoft @MaryGao do you have any further concern?